app/javascript/components/visualization/controls/ScatterPlotLegend.jsx

```diff
@@ -60,16 +60,38 @@ function LegendEntry({
 
   // clicking the label will either hide the trace, or pop up a color picker
   const entryClickFunction = showColorControls ? () => setShowColorPicker(true) : toggleSelection
+
+  let scatterHoverHandle
+  let scatterHoverId
+  // do not trigger `setActiveTraceLabel()` immediately to prevent triggers on scroll
+  const scatterHoverHandler = (label, currentSelectCheck) => {
+    clearTimeout(scatterHoverHandle)
+    scatterHoverHandle = setTimeout(() => {
+      if (scatterHoverHandle && currentSelectCheck()) {
```

Comment:
Could you add a code comment here briefly explaining currentSelectCheck?  

I vaguely grok it after looking at the code for a while, but a brief comment that touches on e.g. why this needs to be a function parameter (i.e. `(label, currentSelectCheck)`) instead of a boolean parameter (e.g. `(label, isTraceCurrentlySelected)`) would make this more maintainable.

Reply:
I don't _think_ there's actually much reason to use the function over boolean, that was what I had seen so I followed the examples, but I switched to boolean param instead and saw no difference. I will double check but likely could switch this to `isTraceCurrentlySelected`